bootstrap/bootstrap.go

```diff
@@ -75,49 +79,71 @@ func translateInterruptToCancel(ctx context.Context, wg *sync.WaitGroup, cancel
 func Run(
 	ctx context.Context,
 	cancel context.CancelFunc,
-	configDir, profileDir, configFileName string,
-	useRegistry bool,
+	configFileName string,
+	commonFlags commandline.CommonFlags,
 	serviceKey string,
-	config interfaces.Configuration,
+	serviceConfig interfaces.Configuration,
 	startupTimer startup.Timer,
 	dic *di.Container,
 	handlers []interfaces.BootstrapHandler) {
 
 	lc := logging.FactoryToStdout(serviceKey)
 	var err error
+	var configClient configuration.Client
 	var registryClient registry.Client
 	var wg sync.WaitGroup
 	translateInterruptToCancel(ctx, &wg, cancel)
 
 	// load configuration from file.
-	if err = configuration.LoadFromFile(configDir, profileDir, configFileName, config); err != nil {
+	if err = config.LoadFromFile(commonFlags.ConfigDirectory(), commonFlags.Profile(), configFileName, serviceConfig); err != nil {
 		fatalError(err, lc)
 	}
 
-	// override file-based configuration with environment variables.
-	bootstrapConfig := config.GetBootstrap()
-	registryInfo, startupInfo := configuration.OverrideFromEnvironment(bootstrapConfig.Registry, bootstrapConfig.Startup)
-	config.SetRegistryInfo(registryInfo)
+	// initialize config provider configuration for URL set in commandline options
+	providerConfig := types.ServiceConfig{}
+	if commonFlags.ConfigProviderUrl() != "" {
+		if err := providerConfig.PopulateFromUrl(commonFlags.ConfigProviderUrl()); err != nil {
+			fatalError(err, lc)
+		}
+	}
 
+	// override file-based configuration with environment variables.
+	bootstrapConfig := serviceConfig.GetBootstrap()
+	providerConfig, startupInfo, err := config.OverrideFromEnvironment(providerConfig, bootstrapConfig.Startup)
+	if err != nil {
+		fatalError(err, lc)
+	}
 	//	Update the startup timer to reflect whatever configuration read, if anything available.
 	startupTimer.UpdateTimer(startupInfo.Duration, startupInfo.Interval)
 
-	// set up registryClient and loggingClient; update configuration from registry if we're using a registry.
-	switch useRegistry {
+	// set up configClient and loggingClient; update configuration from configuration provider if we're using a provider.
+	switch providerConfig.Host != "" {
 	case true:
-		registryClient, err = configuration.UpdateFromRegistry(ctx, startupTimer, config, lc, serviceKey)
+		configClient, err = config.UpdateFromConfigProvider(ctx, startupTimer, providerConfig, serviceConfig, lc, serviceKey)
 		if err != nil {
 			fatalError(err, lc)
 		}
-		lc = logging.FactoryFromConfiguration(serviceKey, config)
-		configuration.ListenForChanges(ctx, &wg, config, lc, registryClient)
+		lc = logging.FactoryFromConfiguration(serviceKey, serviceConfig)
+		config.ListenForChanges(ctx, &wg, serviceConfig, lc, configClient)
+		lc.Info(fmt.Sprintf("Loaded configuration from %s", providerConfig.GetUrl()))
```

Comment:
Yes, I do think that would be useful.

Reply:
I don't think it's beyond the scope of this issue.  You're adding logging within your new implementation to capture where configuration is coming from.  Be consistent and log all the things configuration-state changing.  This would include environment variable overrides.  Do it all or do none of it.  To do only some of it (as is the current case) provides an incomplete and potentially misleading picture.

Reply:
@michaelestrin , I am good with logging the 3 env overrides that are in this code. The remaining are currently in config seed. The logging for those can be added when config seed is removed and the self seed code is added to bootstrap. This work for you?

Reply:
Yes.  Thank you!